#!/usr/bin/env python
PACKAGE = "rc_silhouettematch_client"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("max_number_of_detected_objects", int_t, 0, "Maximum number of detected objects", 10, 1, 20)
gen.add("edge_sensitivity", double_t, 0, "Sensitivity of the edge detector", 0.6, 0.1, 1)
gen.add("match_max_distance", double_t, 0, "Maximum allowed distance in pixels between the template and the detected edges in the image", 2.5, 0, 10)
gen.add("match_percentile", double_t, 0, "Percentage of template pixels that must be within the maximum distance to successfully match the template", 0.85, 0.7, 1)
gen.add("quality", str_t, 0, "Detection quality (H(igh), M(edium), or L(ow))", "H")

gen.add("publish_vis", bool_t, 0, "Whether TF and markers should be published", True)


exit(gen.generate(PACKAGE, "rc_silhouettematch_client", "SilhouetteMatch"))
